pkg/proxy/ipvs/proxier.go

```diff
@@ -614,10 +614,19 @@ func (em proxyEndpointsMap) unmerge(other proxyEndpointsMap) {
 }
 
 // CanUseIPVSProxier returns true if we can use the ipvs Proxier.
-// This is determined by checking if all the required kernel modules are loaded. It may
+// This is determined by checking if all the required kernel modules can be loaded. It may
 // return an error if it fails to get the kernel modules information without error, in which
 // case it will also return false.
 func CanUseIPVSProxier() (bool, error) {
+	// Try to load IPVS required kernel modules using modprobe
+	for _, kmod := range ipvsModules {
+		err := utilexec.New().Command("modprobe", "--", kmod).Run()
```

Comment:
PR comment says dry-run, this does not ?

Reply:
Does this work if kube-proxy is in a container?

@MrHohn

Reply:
I believe this depends on the works on https://github.com/kubernetes/kubernetes/pull/52003.

Reply:
> PR comment says dry-run, this does not ?

I did a simple test and found `modprobe --dry-run ${kmod}` == `modprobe -- ${kmod}`

@vfreex right?

Reply:
> Does this work if kube-proxy is in a container?

this updated logic does no harm when kube-proxy is in a container.

Reply:
How is `--` the same as `--dry-run` ?  That's not in the docs and doesn't make much sense.

Reply:
@thockin Sorry I removed the `--dry-run` option in latter commits.
The latest logic is, try to running `modprobe` to load required kernel modules, and ignore the error if `modprobe` fails. Then check if all those modules are already loaded in the kernel.
Successful running `modprobe` in a container requires `/lib/moudules` on node is mounted to the container. If not doing so, `modprobe` will fail but the error will be ignored.

Reply:
`--` is different from `--dry-run`. `--dry-run` only checks the existence of the modules without actually loading it.
`--` is only an explicit separator to prevent from parsing options: `modprobe -- ${kmod}` is the same as `modprobe ${kmod}`.